ibis/expr/window.py

```diff
@@ -83,16 +83,34 @@ def _validate_frame(self):
             )
         elif preceding_tuple:
             start, end = self.preceding
+            if end is None:
+                raise com.IbisInputError("preceding end point cannot be None")
             if start is None:
-                assert end >= 0
-            else:
-                assert start > end
+                if end < 0:
+                    raise com.IbisInputError(
+                        "preceding end point must be non-negative"
+                    )
+            else:  # start is not None
+                if start <= end:
+                    raise com.IbisInputError(
+                        "preceding start must be greater than preceding end"
+                    )
         elif following_tuple:
             start, end = self.following
+            if start is None:
+                raise com.IbisInputError(
+                    "following start point cannot be None"
+                )
             if end is None:
```

Comment:
Same as above comment.
```suggestion
             if start < 0:
                 raise com.IbisInputError (
                     "following end point must be non-negative"
                 )
             else if end is not None:
```